@xen-orchestra/web-core/lib/components/ui/circle-progress-bar/UiCircleProgressBar.vue

```diff
@@ -0,0 +1,152 @@
+<!-- v3 -->
+<template>
+  <div class="progress-circle-container">
+    <svg
+      :width="circleSize"
+      :height="circleSize"
+      :viewBox="`0 0 ${circleSize} ${circleSize}`"
+      xmlns="http://www.w3.org/2000/svg"
+      class="progress-circle"
+    >
+      <circle
+        :r="radius"
+        :cx="circleSize / 2"
+        :cy="circleSize / 2"
+        fill="transparent"
+        class="progress-circle-background"
+      />
+      <circle
+        :r="radius"
+        :cx="circleSize / 2"
+        :cy="circleSize / 2"
+        fill="transparent"
+        class="progress-circle-foreground progress-circle-fill"
+      />
+    </svg>
+    <div v-if="size !== 'extra-small'" class="progress-circle-overlay">
+      <VtsIcon v-if="isComplete" :icon="icon" class="progress-circle-icon" :accent="iconAccent" />
+      <span v-else class="progress-circle-text" :class="fontClass">{{ percentValue }}</span>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import VtsIcon from '@core/components/icon/VtsIcon.vue'
+import { faCheck, faExclamation } from '@fortawesome/free-solid-svg-icons'
+import { computed } from 'vue'
+
+const { accent, size, value, maxValue } = defineProps<{
+  accent: ProgressCircleAccent
+  size: ProgressCircleSize
+  value: number
+  maxValue: number
+}>()
+
+type ProgressCircleAccent = 'info' | 'success' | 'warning' | 'danger'
+type ProgressCircleSize = 'extra-small' | 'small' | 'medium' | 'large'
+
+const circleSizeMap = {
+  'extra-small': 16,
+  small: 40,
+  medium: 64,
+  large: 164,
+}
+
+const iconSizeMap = {
+  'extra-small': '',
+  small: '1.6rem',
+  medium: '3.2rem',
+  large: '4.8rem',
+}
+
+const strokeWidthMap = {
+  'extra-small': 2,
+  small: 4,
+  medium: 6,
+  large: 16,
+}
+
+const fontClasses = {
+  'extra-small': '',
+  small: 'typo-body-bold-small',
+  medium: 'typo-h5',
+  large: 'typo-h3',
+}
+
+const circleSize = computed(() => circleSizeMap[size])
+const fontClass = computed(() => fontClasses[size])
+const iconSize = computed(() => iconSizeMap[size])
+const strokeWidth = computed(() => strokeWidthMap[size])
+
+const radius = computed(() => (circleSize.value - strokeWidth.value) / 2)
+const circumference = computed(() => 2 * Math.PI * radius.value)
+
+const isComplete = computed(() => value >= maxValue)
+
+const strokeColor = computed(() =>
+  size === 'extra-small'
+    ? `var(--color-${accent}-item-base)`
+    : isComplete.value && (accent === 'info' || accent === 'success')
+      ? 'var(--color-success-item-base)'
+      : `var(--color-${accent}-item-base)`
+)
```

Comment:
I think we could have better readability if we extract a config object, like above:
```suggestion
const strokeColorMap = {
  info: 'var(--color-info-item-base)',
  success: 'var(--color-success-item-base)',
  warning: 'var(--color-warning-item-base)',
  danger: 'var(--color-danger-item-base)',
}

const strokeColor = computed(() => {
  if (isComplete.value && (accent === 'info' || accent === 'success')) {
    return strokeColorMap.success
  }

  return strokeColorMap[accent]
})
```